knimin/lib/tests/test_data_access.py

```diff
@@ -1451,6 +1457,79 @@ def _basic_test_steps_for_normalized_shotgun_plate(self, before, after,
 
             self.assertEqual(obs[k], exp_rnsp[k])
 
+    def test_generate_i5_i7_indexes(self):
+        # these tests will depend on what's in the db, which are added
+        # just below -- Add shotgun_index valid values
+
+        # generate 10 and check the values are just fine
+        obs = db.generate_i5_i7_indexes('iTru', 10)
+        exp = [('iTru5_01_A', 'iTru7_101_01'), ('iTru5_01_A', 'iTru7_101_02'),
```

Comment:
We will want to increment both the i5 and the i7 index with each sample. This has to do with the algorithm the sequencers use for detecting clusters, which depends on a certain amount of base heterogeneity.